project/api/models.py

```diff
@@ -109,24 +111,28 @@ class Meta:
     history = HistoricalRecords()
 
     def clean(self, *args, **kwargs):
+        """Ensure that the input data is valid."""
```

Comment:
Ensure that our geometry is a polygon and our dates do not intersect with those of another Territory of the same Entity.